Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardInputField.swift

```diff
@@ -0,0 +1,195 @@
+
+    @Binding var text: String
+    let inputType: CreditCardInputType
+    let inputViewModel: CreditCardInputViewModel
+    var showError: Bool = false
+
+    // Theming
+    @Environment(\.themeType) var themeVal
+    @State var errorColor: Color = .clear
+    @State var titleColor: Color = .clear
+    @State var textFieldColor: Color = .clear
+    @State var backgroundColor: Color = .clear
+
+    init(inputType: CreditCardInputType,
+         text: Binding<String>,
+         showError: Bool,
+         inputViewModel: CreditCardInputViewModel
+    ) {
+        self.inputType = inputType
+        self._text = text
+        self.showError = showError
+        self.inputViewModel = inputViewModel
+
+        switch self.inputType {
+        case .name:
+            fieldHeadline = .CreditCard.EditCard.NameOnCardTitle
+            errorString = .CreditCard.ErrorState.NameOnCardSublabel
+            delimiterCharacter = nil
+            userInputLimit = 100
+            formattedTextLimit = 100
+            keyboardType = .alphabet
+        case .number:
+            fieldHeadline = .CreditCard.EditCard.CardNumberTitle
+            errorString = .CreditCard.ErrorState.CardNumberSublabel
+            delimiterCharacter = "-"
+            userInputLimit = 19
+            formattedTextLimit = 23
+            keyboardType = .numberPad
+        case .expiration:
+            fieldHeadline = .CreditCard.EditCard.CardExpirationDateTitle
+            errorString = .CreditCard.ErrorState.CardExpirationDateSublabel
+            delimiterCharacter = " / "
+            userInputLimit = 4
+            formattedTextLimit = 7
+            keyboardType = .numberPad
+        }
+    }
+
+    var body: some View {
+        ZStack {
+            VStack(alignment: .leading, spacing: 0) {
+                provideInputField()
+
+                if showError {
+                    errorViewWith(errorString: errorString)
+                }
+            }
+            .background(backgroundColor.edgesIgnoringSafeArea(.bottom))
+        }
+        .padding(.leading, 20)
+        .onAppear {
+            applyTheme(theme: themeVal.theme)
+        }
+        .onChange(of: themeVal) { val in
+            applyTheme(theme: val.theme)
+        }
+    }
+
+    func applyTheme(theme: Theme) {
+        let color = theme.colors
+        errorColor = Color(color.textWarning)
+        titleColor = Color(color.textPrimary)
+        textFieldColor = Color(color.textSecondary)
+        backgroundColor = Color(color.layer2)
+    }
+
+    @ViewBuilder private func provideInputField() -> some View {
+        Text(fieldHeadline)
+            .font(.subheadline)
+            .foregroundColor(titleColor)
+        TextField("", text: $text)
+            .font(.body)
+            .padding(.top, 7.5)
+            .foregroundColor(textFieldColor)
+            .keyboardType(keyboardType)
+            .onChange(of: text) { [oldValue = text] newValue in
+                switch inputType {
```

Comment:
Since we're handling the newest user input and altering the whole `text`, much of it seems hard to move. What could be moved has already been moved, like in expiration's case.

Reply:
Would be nice to move this into view model, in theory your whole switch case can move into view model inside a method and on change simply calls the method which is inside view mode 🤞🏼

Reply:
I see what you mean. I'll try it out!